sphinx/util/rst.py

```diff
@@ -54,7 +54,7 @@ def charwidth(char: str, widechars: str) -> int:
 def heading(env: Environment, text: str, level: int = 1) -> str:
     """Create a heading for *level*."""
     assert level <= 3
-    width = textwidth(text, WIDECHARS[env.language])
+    width = textwidth(text, WIDECHARS[env.language])  # type: ignore[attr-defined]
```

Comment:
This is because of 

https://github.com/sphinx-doc/sphinx/blob/fde43ed8b4367d6f32cbeb7917e76f4be238a190/sphinx/util/template.py#L94-L105

where `language` is an extended attribute. But fair enough, it's better if I double check this one.

Reply:
It's due to 

https://github.com/sphinx-doc/sphinx/blob/fde43ed8b4367d6f32cbeb7917e76f4be238a190/sphinx/util/template.py#L94-L105

which automatically injects the attribute. I'll change it for something safer.